devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -500,13 +504,16 @@ private void handleAutoCompile() throws MojoExecutionException {
                 prepareTestsNeeded = false;
             }
 
-            if (POST_COMPILE_PHASES.contains(goal)) {
+            if (goal.endsWith("compile")) {
```

Comment:
This could satisfy some custom `xxxxcompile` goal. Was there a reason to change the previous way of checking it?

Reply:
It was to avoid having a set with one value as other task are checked with the POST_PROCESS_CLASSES_PHASES.
You are right, compile can be conflicting. I will convert that to equals

Reply:
A set with a single value will do just fine e.g. `Collections.singleton(x)`.

Reply:
fixed